crates/stages/src/stages/bodies.rs

```diff
@@ -51,33 +48,63 @@ use tracing::*;
 #[derive(Debug)]
 pub struct BodyStage<D: BodyDownloader> {
     /// The body downloader.
-    pub downloader: D,
-    /// The consensus engine.
-    pub consensus: Arc<dyn Consensus>,
+    downloader: D,
+    /// Block response buffer.
+    buffer: Vec<BlockResponse>,
```

Comment:
these could probably be an Option, right?
with the assumption that calling execute before ready can panic

but this is fine